### ISSUE TITLE
We have too many wine editions

### ISSUE THREAD

**Issue by abbradar** [NixOS/nixpkgs#26041]
## Issue description

Disclosure: I need a configuration which is not built on Hydra (full+staging) so I have an interest in seeing it supported.

Currently we have:

1. wineMinimal
2. wine
3. wineFull
4. wineStable
5. wineUnstable
6. wineStaging

`wineMinimal`, `wine` and `wineFull` differ in supported features while `wineStable`, `wineUnstable` and `wineStaging` use `wine` feature set while following different release branches. This seems both excessive and not covering enough:

1. People who want to have the latest and greatest features run `wineStaging`. `wine` resolves to the stable channel by default. Is `wineUnstable` actually used much?
2. People who try to run various applications may just run `wineFull` to avoid thinking about what's supported (for example we only have `gtkSupport` and `gstreamerSupport` in `wineFull` while the former is needed for GTK theming (yay, Windows apps are no longer ugly!) and latter is wanted for [a range of games](https://bugs.winehq.org/show_bug.cgi?id=9127)). Default `wine` is a middle ground. What case is covered by `wineMinimal`?

IMHO we should just make `wine = wineFull` and drop current `wine` and `wineMinimal`. I don't think `wine`'s closure size is of such a big importance to split it into three different feature sets because in my mind it belongs (subjectively) to "big desktop apps". I'd also stop building `wineUnstable` but I'm less keen on that. This way we'll just have full-featured `wineStable`, maybe `wineUnstable` and `wineStaging` (that's what I'm interested in personally) built on Hydra; it's both less things to build and covers new cases.

To point it out again it's very subjective, so let's discuss this. 

cc @bendlas @avnik @FRidh 

Closure sizes (staging for no particular reason):
1. `wineMinimal.override { wineRelease = "staging"; }` 668M
2. `wine.override { wineRelease = "staging"; }`: 945M
3. `wineFull.override { wineRelease = "staging"; }`: 1.3G

## Technical details

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`) 9ab21a88b2ecd02c53fe2d8cdcbf864c737d90bf

**Comment by Mic92** [comment]
we probably also want to enable 64-bit by default on x86_64. The 64-bit variant can also run 32-bit application: https://wiki.archlinux.org/index.php/Wine

**Comment by bendlas** [comment]
- `wineFull` is the kitchen sink for everything that can be enabled in wine
  I have no issue with making this the default, but watchers of the closure size might.
- `wineMinimal` is the smallest configuration that allows wine to compile
  I have also no issue with making this the default, but dependents of wine might.
- `wine` is the configuration we had before I factored out the config flags
  I kept it, because I didn't want to break any dependents while maintaining closure size. I haven't evaluated whether `wine` is a good balance between functionality and closure size. I thought it might be, as it has been the default for a while.
- `wine64` can, in fact, not run 32bit code
  For this, a combined variant called `wineWow` is needed with both binaries. This, in effect, doubles the closure size.
  - @edolstra has already reverted my patch making `wineWow` the default on 64bit, because of closure size
    And for good reason too: win64 support in wine is really crappy and most win64 apps won't work anyway
- `wineUnstable` and `wineStaging` are separated in the wine developer community with an explicit remark that not all `staging` patches are `unstable` candidates. I don't know the reasons for this separation, but I'd also rather not get in the middle of that discussion.

All that said, I'm all for cutting down on `top-level` wine derivations, if we can. Maybe a `winePackages` cellar is justified?

**Comment by abbradar** [comment]
Oh, I forgot to mention: I checked out if there are any packages that use `wine` -- there are some that do (`airwave`, pipelight`, `playonlinux`) but no that use `wineMinimal`.

**Comment by bendlas** [comment]
Yeah, I created `wineMinimal` as the baseline to incrementally enable configuration flags on.

**Comment by abbradar** [comment]
About the cellar: it sounds good (at least for allowing people to use shorthands instead of `.override`s when they need a specific configuration). Still, the question of what do we want in the top level persists...

**Comment by bendlas** [comment]
In that case: Just `wine` and `winePackages`.

**Comment by abbradar** [comment]
Good idea but let me rephrase the question then: what do we want to build on Hydra? `wineStaging` is useful for running games so it should probably be built in some configuration. There's also question of what feature sets do we want to build. Wine is a big beast so leaving users without prebuilt binaries is bad.

**Comment by bendlas** [comment]
I see I'd propose two hydra builds:

- `sysadminsWine`: The minimal configuration to support our dependent packages and run boring apps like MS Word reasonably well.
- `gamersWine`: `wineFull` + `staging` + `wineWow`

(please don't call any actual packages "sysadmin swine" or "gamer swine" ;-) )

**Comment by abbradar** [comment]
Sounds good to me! We could just call those `wine` and `wineStaging` respectively to avoid swine jokes :)

Let's wait for other opinions to arrive for a few days; I can prepare a patch later.

**Comment by danbst** [comment]
I did use `wineUnstable` because didn't know `wineStaging` is the most recent version.

**Comment by avnik** [comment]
On Thu, May 25, 2017 at 01:08:13AM -0700, Danylo Hlynskyi wrote:
> I did use `wineUnstable` because didn't know `wineStaging` is the most recent version.
> 

wineStaging is set of experimental patches, some of them unmerged for
years.

**Comment by abbradar** [comment]
See #26152.
